source/PlayState.hx

```diff
@@ -49,7 +62,37 @@ class PlayState extends FlxState
 	override public function update(elapsed:Float):Void
 	{
 		super.update(elapsed);
-		hud.update(elapsed);
+
 		FlxG.collide(map, player);
+
+		// If platform and player are not touching, allow offset to be overwritten
+		if (!FlxG.overlap(player, platform)) {
+			platform.sticky = false;
+		}
+
+		if (FlxG.collide(player, platform)) {
+			// Only set offset value if touching platform for "first" time
+			player.acceleration.y = 8000;
+			if (!platform.sticky) {
+				platform.sticky = true;
+				platform.offsetX = player.x - platform.x;
+			}
+			
+			// Multiply velocity by elapsed to get the player's movement each frame.
+			platform.offsetX += player.velocity.x * elapsed;
+			
+			// Allow player to drop through platform if down key pressed
+			if (FlxG.keys.anyPressed([FlxKey.DOWN])) {
+				player.y = platform.y;
+			}
+
+			// Update player x position
+			player.x = platform.x + platform.offsetX;
+		}
+		else {
+			player.acceleration.y = 600;
+		}
```

Comment:
This whole section of player-platform collision logic should be moved out of playstate, if possible. 
Calls to collide and overlap will still be here, but the rest of this can be moved to functions inside your Platforms class.